irida_staramr_results/irida_api.py

```diff
@@ -112,112 +110,169 @@ def _create_session(self):
+            error_txt = "No analysis found in project."
+            logging.error(error_txt)
+            raise exceptions.IridaKeyError(error_txt)
+
+        all_analysis_results = self._get_all_analysis_results(project_analysis_submissions)
+
+        # Filter AMR Detection results
+        amr_analysis_results = [analysis_result for analysis_result in all_analysis_results if
+                                self._is_type_amr(analysis_result)]
+
+        return amr_analysis_results
+
+    def _get_analysis_submissions_from_projects(self, project_id):
+        """
+        Returns an array of all analyses for a given project
+        :param project_id:
         :return project_analysis_list:
         """
-        project_analysis_list = []
+        try:
+            project_url = self._get_link(self.base_url + f"/api/projects/{project_id}", "self")
+            analysis_submissions_url = self._get_link(project_url, "project/analyses")
+        except StopIteration:
+            logging.error(f"The given project ID doesn't exist: {project_id}")
+            raise exceptions.IridaResourceError("The given project ID doesn't exist", project_id)
 
-        project = self.get_project_from_id(_id).json()
+        response = self._get_resource(analysis_submissions_url)
+        analysis_submissions = response.json()["resource"]["resources"]
 
-        project_analyses_url = self._get_href_from_links("project/analyses", project["resource"]["links"])
+        return analysis_submissions
 
-        if project_analyses_url is None:
-            print("No 'project/analyses' link relation in project object.")
-        else:
-            project_analysis = {}  # analysis object
+    def _get_all_analysis_results(self, project_analysis_submissions):
 
-            project_analyses_response = self._get_resource(project_analyses_url).json()
+        analysis_result_list = []
 
-            # check if resources array is not empty
-            if len(project_analyses_response["resource"]["resources"]) <= 0:
-                print("No analysis in this project")
-            else:
-                for analysis in project_analyses_response["resource"]["resources"]:
-                    # create the analysis object, overrides existing information if it exists.
-                    project_analysis["name"] = analysis["name"]
-                    project_analysis["identifier"] = analysis["identifier"]
-                    project_analysis["workflow_id"] = analysis["workflowId"]
-                    project_analysis["links"] = analysis["links"]
+        for analysis_submission in project_analysis_submissions:
+
+            analysis_submission_id = analysis_submission["identifier"]
 
-                    # add to the list of analysis, by deep copy
-                    project_analysis_list.append(project_analysis.copy())
+            try:
+                analysis_results_url = self._get_link(self.base_url + f"/api/analysisSubmissions/{analysis_submission_id}",
```

Comment:
Same as above, if possible we should break this into multiple `_get_link` calls using `target_dict`, it will be safely handle the calls.

The additional benefit is because `_get_link` is querying for the identifiers (in this case `analysis_submission_id`) it handles the `KeyError` for us, and generates the `IridaKeyError` for us, raising it to the function caller so we don't even need to have the `try/except` block

Reply:
Done. I broke it down to multiple _get_link calls.

However, on your next point, I kept the `try/except` block because it is possible for an analysis submission to not have the`"ref": "analysis"` and I needed a way to catch it and handle it by ignoring it. I change the exception type from `KeyError` to `IridaKeyError` to be more specific.